webapp/client/src/stories/FormFieldIntegerInput.stories.js

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+
+import FormFieldIntegerInput from "../components/FormFieldIntegerInput";
+import StepForm from "../components/StepForm";
+import { Default as StepFormDefault } from "./StepForm.stories";
+
+export default {
+  title: "Form Fields/IntegerInput",
+  component: FormFieldIntegerInput,
+};
+
+function Template(args) {
+  return (
+    <StepForm {...StepFormDefault.args}>
+      <FormFieldIntegerInput {...args} />
+    </StepForm>
+  );
+}
+
+function TemplateLong(args) {
+  const style = {
+    height: 1000,
+  };
+  return (
+    <StepForm {...StepFormDefault.args}>
+      <FormFieldIntegerInput {...args} />
+      <div style={style} />
+    </StepForm>
+  );
+}
+
+export const Default = Template.bind({});
+export const WithValue = Template.bind({});
+export const WithErrors = Template.bind({});
+export const WithDetails = Template.bind({});
+export const WithAllLabelExtras = Template.bind({});
+export const WithMaxWidth = Template.bind({});
+export const WithMaxLength = Template.bind({});
+export const WithMaxLengthAndWidth = Template.bind({});
+
+export const OnALongPage = TemplateLong.bind({});
+
+Default.args = {
+  fieldId: "number-input",
+  fieldName: "number-input",
+  label: {
+    text: "Grad der Behinderung",
+  },
+  errors: [],
+  value: "",
+};
+
+WithValue.args = {
+  ...Default.args,
+  value: "20",
+};
+
+WithErrors.args = {
+  ...Default.args,
+  errors: ["Dieses Feld wird benötigt"],
+};
+
+WithDetails.args = {
+  ...Default.args,
+  details: {
+    title: "Was bedeutet das?",
+    text: "Erklärungs-Platzhalter",
+  },
+};
+
+WithAllLabelExtras.args = {
+  ...WithDetails.args,
+  label: {
+    text: "Grad der Behinderung",
+    showOptionalTag: true,
+    help: "Das ist ein Hilfetext. Hilft er dir?",
+    exampleInput: "ab 20",
+  },
+};
+
+WithMaxWidth.args = {
+  ...Default.args,
+  maxWidth: 4,
+};
+
+WithMaxLength.args = {
+  ...Default.args,
+  maxLength: 4,
+};
+
+WithMaxLengthAndWidth.args = {
+  ...Default.args,
+  maxLength: 4,
+  maxWidth: 4,
+};
```

Comment:
This feels overkill to me? I don't think we should enumerate every possible state (especially since you can modify these values in Storybook very easily). Instead, let's focus on **interesting** states, maybe: a bare field, one with errors, one with a fully decked out details component.

Reply:
Mh, not sure I agree with you here. I spoke to Design and they explicitly requested some stories that could have "just" been configured in Storybook. So I see this more as a showcase of everything that is possible and can/should be thought about when designing an input.
Plus: Does it really hurt?

Reply:
Ok that's a fair need. I do think having to enumerate all these different variants on every single component _does_ hurt (as does having to backfill something we will inevitably forget). It's code we have to write and maintain. I wonder if it wouldn't be better to configure controls for the Story args with https://storybook.js.org/docs/react/essentials/controls and teach the designers to use Storyboard? That way we'd get ALL enumerations accessible to them. Check out the link and let me know what you think!

Reply:
Yes, I think you're right. And good to know that it is possible to do way more with the stories than just declare them. 👌 
However, I don't see the customization that we'd need in my opinion: pre-showing the object that you can fill. E.g. for the details: if no details are set, there is just an option for "details: {}" and no indication that the keys put in there should be `text` and `title`. So could this maybe be a good rule for now? "Add a story for things that need object input and for interesting states"? 
For this specific case that would mean: completely omit WithValue, WithDetails, WithMaxWidth, WithMaxLength and WithMaxLengthAndWidth and just keep WithAllLabelExtras.